src/fidesops/service/connectors/saas/authenticated_client.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from typing import Optional
+
+from requests import PreparedRequest, Request, Response, Session
+
+from fidesops.common_exceptions import ClientUnsuccessfulException, ConnectionException
+from fidesops.core.config import config
+from fidesops.models.connectionconfig import ConnectionConfig
+from fidesops.schemas.saas.saas_config import ClientConfig
+from fidesops.schemas.saas.shared_schemas import SaaSRequestParams
+
+logger = logging.getLogger(__name__)
+
+
+class AuthenticatedClient:
+    """
+    A helper class to build authenticated HTTP requests based on
+    authentication and parameter configurations. Optionally allows
+    a request client config to override the root client config.
+    """
+
+    def __init__(
+        self,
+        uri: str,
+        configuration: ConnectionConfig,
+        request_client_config: ClientConfig = None,
```

Comment:
Moved `AuthenticatedClient` to its own file but the change getting drowned out by the move is the addition of an optional `request_client_config` to override the root `client_config`. I know we do something similar within SaaSConnector but I wanted to be able to do this for a standalone `AuthenticatedClient`.